src/libs/ajax/Catalog.ts

```diff
@@ -0,0 +1,169 @@
+import * as _ from 'lodash/fp'
+import { authOpts, fetchCatalog, jsonBody } from 'src/libs/ajax/ajax-common'
+
+
+interface DatasetTableListEntry {
+  name: string
+  hasData?: boolean
+}
+
+interface DatasetTableListResponse {
+  tables: DatasetTableListEntry[]
+}
+
+interface Column {
+  name?: string
+}
+
+interface DatasetTableResponse {
+  columns: Column[]
+  rows: any[]
+}
+
+
+type DataUsePermission =
+    'DUO:0000007' |
```

Comment:
What is the value of using an enum type here? The UI doesn't ever generate this data; it's only generated by the service, so it doesn't seem like it would be useful for writing UI code. The downside I see is that it will need to be updated when the schema changes, so it seems like a downside with no benefit to me.